clients/privacy-center/components/modals/consent-request-modal/ConsentRequestForm.tsx

```diff
@@ -44,6 +44,7 @@ const useConsentRequestForm = ({
   const identityInputs =
     config.consent?.button.identity_inputs ?? defaultIdentityInput;
   const toast = useToast();
+  const cookie = useMemo(() => getOrCreateFidesCookie(), []);
```

Comment:
this will create the cookie as soon as the form opens rather than on form submission, is that okay?

Reply:
Yeah - this doesn't actually _save_ the cookie yet, it just creates one. That might be surprising behaviour though and worth second guessing...

Basically the behaviour of `getOrCreateFidesCookie` is:
1) Read the cookies to see if one exists - if so, parse it and return as a JS object
2) If one doesn't exist, create one in memory and return it (but don't save to the browser)

I _like_ that pattern, it's very common for ORM-style resources to be things like `const user = new User(); user.save()`, but I'm wondering if we can rename the methods to make that clear...

Reply:
Alternatively, one *more* way to do this would be to have the privacy center responsible for the initialization logic, so something more like:
1. Read the cookie from browser and either:
  a. If the cookie exists, use the UUID from the cookie
  b. If the cookie doesn't exist, generate a new UUID
2. When the consent form saves successfully, either:

Reply:
I think I prefer encapsulating that UUID initialization logic in the cookie code (like I have it now) but I would like the API to be more self-documenting... right now `getOrCreate` is a pure function without side effects, but I think the `...OrCreate` part makes you think it saves.

Any ideas for a better function name...? `getExistingCookieOrReturnUnsavedDefaultCookie()`?

I think

Reply:
ah, got it. I think I was still thinking about the previous way, so `create` to me meant "create/set the cookie". but if I hadn't known the previous way, maybe I wouldn't have thought that way and that func would've made total sense!

`getExistingCookieOrReturnUnsavedDefaultCookie()` certainly works, though I wonder if `getOrCreateFidesCookieValue` would work as well? creating a value doesn't imply setting IMO, and also has the benefit of knowing we aren't actually getting a cookie string back—it's already parsed and all that. maybe `getOrCreateFidesCookieObject`? 🤔

Reply:
maybe `getOrMake` ? like how we have `makeConsentDefaults`

Reply:
I did `getOrMakeFidesCookie` 👍